scalanet/src/io/iohk/scalanet/crypto/CryptoUtils.scala

```diff
@@ -122,7 +122,8 @@ private[scalanet] object CryptoUtils {
       random: SecureRandom,
       extensions: List[Extension],
       beforeDate: Date,
-      afterDate: Date
+      afterDate: Date,
+      signatureScheme: SignatureScheme
```

Comment:
Yep I mostly added it here to test negative case.

Original spec allows many possible combinations signatures and used hashes, it only requires that:
- hashes has at least 256bit
- RSA is avoided (but not forbidden)

I just restricted it here a little more as that way:
- it is easier to validate it
- I treat `scalanet` mostly as our project network layer not a generic network library as libp2p
- we already restrict some parts of the spec for our case, for example we only support secp256k1 keys in extension, where is original spec allows for 4 different kinds of keys.

My thinking goes, for now let's make everything as tailored to our use case as possible, and if we ever need to integrate with some software that uses libp2p tls connections, then we will extend our implementation to be as close to spec as possible.

And as far as default values, I try to avoid using them as they bit me several times in the past.